### PR TITLE
`filter` is an implementation detail

### PR DESCRIPTION
**Pull request by ashtum**

### CODE REVIEW

File: include/boost/http_proto/detail/filter.hpp [hunk 1]
```diff
@@ -21,14 +20,16 @@
...
 namespace boost {
 namespace http_proto {
+namespace detail {
 
 /** A filter for buffers
 */
-class BOOST_HTTP_PROTO_DECL
-    filter
-    : public buffered_base
+class filter
```
Comment (vinniefalco) [inline include/boost/http_proto/detail/filter.hpp:27]: this type is still publicly visible. can't we just forward-declare it and only declare it in the .cpp files?

Comment (vinniefalco) [reply include/boost/http_proto/detail/filter.hpp:27]: lets do it

File: src_zlib/service/zlib_service.cpp [hunk 2]
```diff
@@ -335,7 +335,7 @@ init(bool use_gzip)
     stream_.avail_in = 0;
 }
 
-filter::results
+http_proto::detail::filter::results
```
Comment (vinniefalco) [inline src_zlib/service/zlib_service.cpp:338]: auto with a trailing return type might make this shorter?

**Comment by vinniefalco** [comment]
I left one comment, otherwise this seems OK.

File: src_zlib/service/zlib_service.cpp (hunk 2 above)
Comment (ashtum) [reply src_zlib/service/zlib_service.cpp:337]: `deflate_filter` is in a different namespace. we need the whole `http_proto::detail::filter::results`.

**Pull request merged**